.github/workflows/migration.yml

```diff
@@ -0,0 +1,55 @@
+name: Database Migration Test
+on:
+  workflow_call:
+jobs:
+    test:
+      name: MySQL -> Postgres Migration
+      runs-on: ubuntu-22.04
+      env:
+          COMPOSE_PROJECT_NAME: ghactions
+          BUILD_IMAGE: mattermost/mattermost-build-server:20230118_golang-1.19.5
+      defaults:
+        run:
+          working-directory: server
+      steps:
+      - name: Checkout mattermost project
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: Run docker compose
+        run: |
+          cd build
+          docker-compose --ansi never run --rm start_dependencies
+          cat ../tests/test-data.ldif | docker-compose --ansi never exec -T openldap bash -c 'ldapadd -x -D "cn=admin,dc=mm,dc=test,dc=com" -w mostest';
+          docker-compose --ansi never exec -T minio sh -c 'mkdir -p /data/mattermost-test';
+          docker-compose --ansi never ps
+      - name: Generate test-data
+        run: |
+          docker run --net ${COMPOSE_PROJECT_NAME}_mm-test \
+            --ulimit nofile=8096:8096 \
+            --env-file=build/dotenv/migration.env \
+            -v $(go env GOCACHE):/go/cache \
+            -e GOCACHE=/go/cache \
+            -v $PWD:/mattermost \
+            -w /mattermost \
+            $BUILD_IMAGE \
+            make test-data
```

Comment:
As you mentioned the key here will be to generate sufficiently diverse data to gain confidence that the migration is stressing most (if not all) tables/columns.

Do we have a ticket to plan improvements on this front? Regardless of its implementation (SQL functions vs server command or else) it would be good to prioritize this sort of work.

Reply:
Makes sense, I think it would be best to generate via the server app, to get a real life data. Created the ticket here: https://mattermost.atlassian.net/browse/MM-53979